taichi/ir/ir.h

```diff
@@ -423,14 +423,21 @@ class StmtField {
 template <typename T>
 class StmtFieldNumeric final : public StmtField {
  private:
+  T *value_ptr;
   T value;
```

Comment:
nit: Consider using `std::variant`?

Reply:
Wow, that's something I haven't think of, I'll definitely try that